lnbits/commands.py

```diff
@@ -59,64 +55,37 @@ def bundle_vendored():
-            if match:
-                version = int(match.group(1))
-                if version > current_versions.get(db_name, 0):
-                    logger.debug(f"running migration {db_name}.{version}")
-                    await migrate(db)
-
-                    if db.schema == None:
-                        await set_migration_version(db, db_name, version)
-                    else:
-                        async with core_db.connect() as conn:
-                            await set_migration_version(conn, db_name, version)
-
     async with core_db.connect() as conn:
         if conn.type == SQLITE:
             exists = await conn.fetchone(
                 "SELECT * FROM sqlite_master WHERE type='table' AND name='dbversions'"
             )
         elif conn.type in {POSTGRES, COCKROACH}:
             exists = await conn.fetchone(
-                "SELECT * FROM information_schema.tables WHERE table_name = 'dbversions'"
+                "SELECT * FROM information_schema.tables WHERE table_schema= 'public' AND table_name = 'dbversions'"
             )
 
         if not exists:
             await core_migrations.m000_create_migrations_table(conn)
 
-        rows = await (await conn.execute("SELECT * FROM dbversions")).fetchall()
-        current_versions = {row["db"]: row["version"] for row in rows}
-        matcher = re.compile(r"^m(\d\d\d)_")
-        db_name = core_migrations.__name__.split(".")[-2]
-        await run_migration(conn, core_migrations, db_name)
+        current_versions = await get_dbversions(conn)
+        core_version = current_versions.get("core", 0)
+        await run_migration(conn, core_migrations, core_version)
 
     for ext in get_valid_extensions():
-        try:
+        current_version = current_versions.get(ext.code, 0)
+        await migrate_extension_database(ext, current_version)
 
-            module_str = (
-                ext.migration_module or f"lnbits.extensions.{ext.code}.migrations"
-            )
-            ext_migrations = importlib.import_module(module_str)
-            ext_db = importlib.import_module(f"lnbits.extensions.{ext.code}").db
-            db_name = ext.db_name or module_str.split(".")[-2]
-        except ImportError:
-            raise ImportError(
-                f"Please make sure that the extension `{ext.code}` has a migrations file."
-            )
+    logger.info("✔️ All migrations done.")
 
-        async with ext_db.connect() as ext_conn:
-            await run_migration(ext_conn, ext_migrations, db_name)
 
-    logger.info("✔️ All migrations done.")
+async def db_versions():
+    async with core_db.connect() as conn:
+        current_versions = await get_dbversions(conn)
+        return current_versions
+
+
+async def load_disabled_extension_list() -> None:
+    """Update list of extensions that have been explicitly disabled"""
+    inactive_extensions = await get_inactive_extensions()
+    settings.lnbits_deactivated_extensions += inactive_extensions
```

Comment:
Might have unforeseen consequences to overwrite the setting here. Maybe a separate variable (or setting?) would be more appropriate.

Reply:
There was indeed an issue that the values of `settings` would be overwritten when an admin user would save the `Settings` from the UI. That was fixed.

In order to make matters more explicit the `Settings` class is now an aggregate of 3 classes:

```py
class Settings(EditableSettings, ReadOnlySettings, TransientSettings):
```

- `EditableSettings` - what the admin user sees in the UI and what gets stored in the DB
- `ReadOnlySettings` - read from the `.env` file, cannot be changed from the code or from the UI
- `TransientSettings`:
```py
class TransientSettings(InstalledExtensionsSettings):
    # Transient Settings:
    #  - are initialized, updated and used at runtime
    #  - are not read from a file or from the `setings` table
    #  - are not persisted in the `settings` table when the settings are updated
    #  - are cleared on server restart
```